### PR TITLE
update comments and method order

### PR DESCRIPTION
**Pull request by yochay**
Mostly update function order to streamline reading the code and added more comments.

### CODE REVIEW

File: samples/javascript_nodejs/02.echobot-with-counter/bot.js [hunk 1]
```diff
@@ -12,15 +15,20 @@ class EchoBot {
...
     constructor(conversationState) {
-        // creates a new state accessor property.see https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors
+        // Creates a new state accessor property.
+        // See https://aka.ms/about-bot-state-accessors to learn more about the bot state and state accessors
         this.countProperty = conversationState.createProperty(TURN_COUNTER_PROPERTY);
         this.conversationState = conversationState;
     }
     /**
+     *
+     * Use onTuen to handle an incoming activity, received from a user, process it, and reply as needed
```
Comment (sgellock) [inline samples/javascript_nodejs/02.echobot-with-counter/bot.js]: typo 'onTuen'

File: samples/javascript_nodejs/02.echobot-with-counter/index.js [hunk 2]
```diff
@@ -38,19 +27,38 @@ try {
...
 });
 
+// Catch-all for any unhandled errors in your bot.
+adapter.onTurnError = async (context, error) => {
+    // This check writes out errors to console log .vs. app insights.
+    console.error(`\n [onTurnError]: ${ error }`);
+    // Send a message to the user
+    context.sendActivity(`Oops. Something went wrong!`);
+    // Clear out state
+    conversationState.clear(context);
```
Comment (daveta) [inline samples/javascript_nodejs/02.echobot-with-counter/index.js]: Hmm. I'm not doing this.  Do we need to do this?

Comment (yochay) [reply samples/javascript_nodejs/02.echobot-with-counter/index.js]: @daveta  yes, we should add the catch-all exception

File: samples/javascript_nodejs/02.echobot-with-counter/bot.js [hunk 3]
```diff
@@ -12,15 +15,20 @@ class EchoBot {
...
      *
      * @param {TurnContext} on turn context object.
      */
     async onTurn(turnContext) {
+        // Handle Message activity type, which is the main activity type for shown within a conversational interface
```
Comment (daveta) [inline samples/javascript_nodejs/02.echobot-with-counter/bot.js]: >type for shown within  [](start = 68, length = 22)

"type for shown within" => "type for display in"

File: samples/javascript_nodejs/02.echobot-with-counter/bot.js [hunk 4]
```diff
@@ -29,7 +37,8 @@ class EchoBot {
             await turnContext.sendActivity(`${ count }: You said "${ turnContext.activity.text }"`);
             // increment and set turn counter.
             await this.countProperty.set(turnContext, count);
-        } else {
+        } else { 
+            // Generic hendler for all other activity types.
```
Comment (daveta) [inline samples/javascript_nodejs/02.echobot-with-counter/bot.js]: > Generic hendler [](start = 14, length = 16)

 Generic hendler => Generic handler

File: samples/javascript_nodejs/13.basic-bot/index.js [hunk 5]
```diff
@@ -35,19 +27,40 @@ try {
...
+    console.error(`\n [onTurnError]: ${ error }`);
+    // Send a message to the user
+    context.sendActivity(`Oops. Something went wrong!`);
+    // Clear out state
+    conversationState.clear(context);
+};
+
 // Define a state store for your bot. See https://aka.ms/about-bot-state to learn more about using MemoryStorage.
 // A bot requires a state store to persist the dialog and user state between messages.
 let conversationState, userState;
```
Comment (vishwacsena) [inline samples/javascript_nodejs/13.basic-bot/index.js]: Can you add this line right after line 76? This is the other issue Andrea is running into with the Basic bot.
// const { BlobStorage } = require('botbuilder-azure');

File: samples/javascript_nodejs/02.echobot-with-counter/index.js [hunk 6]
```diff
@@ -76,23 +84,24 @@ conversationState = new ConversationState(memoryStorage);
 // });
 // conversationState = new ConversationState(blobStorage);
 
+// This bot's main dialog.
+const { EchoBot } = require('./bot');
```
[the remark this answers is not in the document]
Comment (yochay) [reply samples/javascript_nodejs/02.echobot-with-counter/index.js]: so sad :(

File: samples/javascript_nodejs/13.basic-bot/index.js [hunk 7]
```diff
@@ -35,19 +27,40 @@ try {
...
+// bot name as defined in .bot file or from runtime
+const BOT_CONFIGURATION = (process.env.NODE_ENV || DEV_ENVIRONMENT);
+
 // Get bot endpoint configuration by service name
+// Bot configuration as defined in .bot file 
 const endpointConfig = botConfig.findServiceByNameOrId(BOT_CONFIGURATION);
 
-// Create adapter. See https://aka.ms/about-bot-adapter to learn more about .bot file its use and bot configuration .
+// Create adapter. 
+// See https://aka.ms/about-bot-adapter to learn more about .bot file its use and bot configuration .
```
[the remark this answers is not in the document]
Comment (yochay) [reply samples/javascript_nodejs/13.basic-bot/index.js]: This is actually the adapter comment - when initializing an adapter

File: samples/javascript_nodejs/13.basic-bot/index.js (hunk 5 above)
Comment (yochay) [reply samples/javascript_nodejs/13.basic-bot/index.js]: done. I added the require statement at the top of the file

**Pull request merged**